KU-11 финальные конфиги Docker, настройка постоянного хранилища БД

KornilovVl opened the pull request:
Настроил постоянное хранилище данных mysql на хосте 
Написал Dockerfile для mysql для устранения ошибок использования одного docker image и docker volume несколькими сервисами 
Изменил настройки пользователя, бд и пароля

KornilovVl committed:
Переименовал папку build для удобства отображения контейнеров в docker desktop
Настроил постоянное хранилище данных mysql на хосте
Написал Dockerfile для mysql для устранения ошибок использования одного docker image и docker volume несколькими сервисами
Изменил настройки пользователя, бд и пароля

tungatarovKl commented on build-messenger/database/Dockerfile:
```diff
@@ -0,0 +1,5 @@
+FROM mysql
```
не очень хорошая идея хранить credentials в dockerfile. Обычно используется 1 докерфайл на прод и тест. В таких случаях не совсем понятно будет где, какие credentials используется.
Обычно такие вещи на проде храниться в более безопасных хранилищах и собираются динамический при деплое. 
Эти credentials можно вынести в .env.example т.к они используется в тестовом окружений. А для прода .env файл собирается отдельно при деплое.

tungatarovKl commented on build-messenger/docker-compose.yml:
```diff
@@ -25,13 +25,12 @@ services:
-            MYSQL_USER: ku1-newsletter-user
-            MYSQL_PASSWORD:
+        volumes:
+            -   ../../ku1-newsletter-messenger-database:/var/lib/mysql
```
можно использовать именованные volume'ы. Как тут https://stackoverflow.com/a/45039609

KornilovVl committed:
Разбил параметры докера на dev и prod среду
Прописал креды базы в конфиг- файле
Укоротил названия контейнеров, сервисов, конфиги БД

KornilovVl committed:
Merge branch 'main' into KU-11

tungatarovKl commented on build/prod/docker-compose.yml:
```diff
@@ -0,0 +1,25 @@
+version: '3'
```
не понял почему удалена dev/docker-compose.yml

KornilovVl replied:
Перенес из build/docker-compose.yml в build/dev/docker-compose.yml . Скорее всего git посчитал файл удаленным а не перенесенным

KornilovVl committed:
Добавил именованный volume для MySQL
Актуализировал README.md

KornilovVl committed:
Merge remote-tracking branch 'origin/KU-11' into KU-11
